src/main/java/net/mcreator/element/types/Item.java

```diff
@@ -205,8 +207,9 @@ public List<StateEntry> getModels() {
 		List<String> builtinProperties = DataListLoader.loadDataList("itemproperties").stream()
 				.filter(e -> e.isSupportedInWorkspace(getModElement().getWorkspace())).map(DataListEntry::getName)
 				.toList();
+		AtomicInteger index = new AtomicInteger();
 
-		states.forEach(state -> {
+		states.stream().sorted(Comparator.comparing(state -> -state.stateMap.size())).forEach(state -> {

```

Comment:
Why do we need to sort them? Also will this impact old states?

Reply:
I can answer the second part. It will impact them so we shouldn't really sort this

Reply:
1. This way the tree would first create branches for states with highest number of properties used so as to match as many property values as possible (first of states in my setup only uses 1 of 2 custom properties defined for the item, and without this change states using 2 properties were ignored if value of that single property was bigger than the lowest threshold specified by those states).
2. This method is used both in item generator YAML and its overrides definition in model file, so if the code gets regenerated, all files should be correctly updated.

Reply:
1.21.1 and before depends on ordering because it select properties by their order so it is not ok to sort them.

> branches for states with highest number of properties used so as to match as many property values as possible

I don't think this would replicate old system where first state matching all predicates is selected

Reply:
One way to solve this is to move sorting to the new method.

Reply:
I agree with that suggestion